firebase-firestore-mixin.html

```diff
@@ -229,6 +225,8 @@
           observesArgs.length === config.observes.length;
 
         if (propArgsReady && observesArgsReady) {
+          this._firestoreUnlisten(name);
+          this.db = this.db || firebase.firestore();
```

Comment:
Made some moves here to get this less likely to happen on first run.

@merlinnot I see what you're saying now about possibly moving `db` to the scope. I think the reason it's used like this is to mimic the way `firebase-database-behavior.html` actually adds `db` to the properties getter https://github.com/firebase/polymerfire/blob/master/firebase-database-behavior.html#L20 not sure if there's benefits to doing that or not here.

Reply:
Ok, I see. I think we might want to rename this property, but I’ll make an issue for this, let’s not do it as a part of this PR.

Original (but broken) approach using `constructor` had a good intention of reusing the same firestore instance. What we should do IMO is:
- create a variable in the scope to cache firestore instance
- create a getter in the class which would assign a value to this variable as a side-effect if it’s undefined, than return it.
- use this variable directly within the class methods to avoid calling the getters logic (performance)
- make sure we initially assign it as late as possible (we might want to extract “assign if undefined”) logic to a helper method to reuse it in the getter and for initial assignment

Does it sound right?

Reply:
I think that makes sense. I'm not quite sure I understand how to achieve what you're talking about in the last two bullets, so I can't say whether I think it's overkill or not. Would this look something like:
```
if (typeof Polymer === 'undefined') {
  throw new Error('Polymer.FirestoreMixin must be imported after Polymer itself.');
}
{
  let DB;

  Polymer.FirestoreMixin = parent =>
    get db() {
      DB = DB || firebase.firestore();
      return DB;
    }
    initDB() {
      DB ? return : this.db;
    }
    _firestoreUpdateBinding(name, ...args) {
      // ...
      if (propArgsReady && observesArgsReady) {
        this._firestoreUnlisten(name);
        initDB();
        // ...
        let ref = DB[config.type](collPath);
        // ...
      }
      // ...
    }
}
```
If so, I'm not completely sure how the "assign if undefined" logic in `initDB` would add any performance benefit over calling `this.db` directly. Also, correct me if I'm wrong, but doesn't this also assume that _every_ instance of this mixin is then attached to the same Firestore remote? I've not spent much time attaching things to scope like this, so this is great learning for me.